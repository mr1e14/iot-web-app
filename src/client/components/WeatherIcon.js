import React from "react";
import matches from "../mediaQuery";
import clearDay from "../img/clear-day.png";
import clearNight from "../img/clear-night.png";
import cloudy from "../img/cloudy.png";
import defaultIcon from "../img/default.png";
import fog from "../img/fog.png";
import partlyCloudDay from "../img/partly-cloudy-day.png";
import partlyCloudyNight from "../img/partly-cloudy-night.png";
import rain from "../img/rain.png";
import sleet from "../img/sleet.png";
import snow from "../img/snow.png";
import thunderstorm from "../img/thunderstorm.png";
import tornado from "../img/tornado.png";
import wind from "../img/wind.png";

const icons = {
  "clear-day": clearDay,
  "clear-night": clearNight,
  cloudy,
  defaultIcon,
  fog,
  "partly-cloudy-day": partlyCloudDay,
  "partly-cloudy-night": partlyCloudyNight,
  rain,
  sleet,
  snow,
  thunderstorm,
  tornado,
  wind
};

const WeatherIcon = props => {
  const icon = props.weatherData ? props.weatherData.icon : "defaultIcon";
  const iconSize = matches("xs") ? "48px" : "64px";
  return (
    <img
      style={{ marginRight: "0.5rem" }}
      src={icons[icon]}
      height={iconSize}
      alt="weather-icon"
      align="middle"
    />
  );
};

export default WeatherIcon;
