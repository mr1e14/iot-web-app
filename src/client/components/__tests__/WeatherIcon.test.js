import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

jest.mock("../mediaQuery");

import matches from "../mediaQuery";
import WeatherIcon from "../WeatherIcon";

const exampleWeatherData = {
  weatherData: {
    icon: "partly-cloudy",
    temperature: "15",
    humidity: "0.75"
  }
};

describe("WeatherIcon", () => {
  let tree;
  let component;
  describe("on xs devices", () => {
    beforeEach(() => {
      matches.mockReturnValue(false);
      component = renderer.create(
        <WeatherIcon weatherData={exampleWeatherData} />
      );
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("on larger devices", () => {
    beforeEach(() => {
      matches.mockReturnValue(true);
      component = renderer.create(
        <WeatherIcon weatherData={exampleWeatherData} />
      );
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when weatherData is not available", () => {
    beforeEach(() => {
      component = renderer.create(<WeatherIcon weatherData={null} />);
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("size changes based on screen size", () => {
    describe("On xs devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(false);
      });
      it("has has a height of 48px", () => {
        const wrapper = mount(<WeatherIcon weatherData={exampleWeatherData} />);
        expect(wrapper.find("img").prop("height")).toEqual("48px");
      });
    });
    describe("On larger devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(true);
      });
      it("has has a height of 64px", () => {
        const wrapper = mount(<WeatherIcon weatherData={exampleWeatherData} />);
        expect(wrapper.find("img").prop("height")).toEqual("64px");
      });
    });
  });
});
