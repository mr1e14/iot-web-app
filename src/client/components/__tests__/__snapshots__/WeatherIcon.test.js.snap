// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIcon on larger devices renders correctly 1`] = `
<img
  align="middle"
  alt="weather-icon"
  height="64px"
  style={
    Object {
      "marginRight": "0.5rem",
    }
  }
/>
`;

exports[`WeatherIcon on xs devices renders correctly 1`] = `
<img
  align="middle"
  alt="weather-icon"
  height="48px"
  style={
    Object {
      "marginRight": "0.5rem",
    }
  }
/>
`;

exports[`WeatherIcon when weatherData is not available renders correctly 1`] = `
<img
  align="middle"
  alt="weather-icon"
  height="64px"
  src="test-file-stub"
  style={
    Object {
      "marginRight": "0.5rem",
    }
  }
/>
`;
